src/main/java/uk/nhs/hee/tis/trainee/forms/interceptor/TraineeIdentityInterceptor.java

```diff
@@ -0,0 +1,87 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.forms.interceptor;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.nhs.hee.tis.trainee.forms.api.util.AuthTokenUtil;
+import uk.nhs.hee.tis.trainee.forms.dto.TraineeIdentity;
+
+/**
+ * An interceptor for creating a {@link TraineeIdentity} from a request.
+ */
+@Slf4j
+public class TraineeIdentityInterceptor implements HandlerInterceptor {
+
+  // Endpoints are a mix of authenticated (public) and unauthenticated (internal), limit
+  // trainee ID verification to LTFT, COJ and FormR endpoints for now.
+  public static final List<String> INTERCEPTOR_API_REGEXPS
+      = List.of("^/api/ltft(/.+)?$", "^/api/formr-part[a|b](s|/.+)?$", "/api/coj");
+
+  private final TraineeIdentity traineeIdentity;
+
+  public TraineeIdentityInterceptor(TraineeIdentity traineeIdentity) {
+    this.traineeIdentity = traineeIdentity;
+  }
+
+  @Override
+  public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
+      Object handler) {
+    String authToken = request.getHeader(HttpHeaders.AUTHORIZATION);
+
+    if (authToken != null) {
+      try {
+        String traineeId = AuthTokenUtil.getTraineeTisId(authToken);
+        traineeIdentity.setTraineeId(traineeId);
+      } catch (IOException e) {
+        log.warn("Unable to extract trainee ID from authorization token.", e);
+      }
+    }
+
+    if (traineeIdentity.getTraineeId() == null && apiRequiresTraineeId(request.getRequestURI())) {
+      response.setStatus(HttpStatus.FORBIDDEN.value());
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Determine if a request URI matches any of the APIs that require a trainee ID.
+   *
+   * @param requestUri The request API to assess.
+   * @return True if it matches any API requiring a trainee ID, otherwise false.
+   */
+  private boolean apiRequiresTraineeId(String requestUri) {
+    for (String regexp : INTERCEPTOR_API_REGEXPS) {
+      if (requestUri.matches(regexp)) {
+        return true;
+      }
+    }
+    return false;
+  }
```

Comment:
This could be configured when registering the handler in `InterceptorConfiguration` using [addPathPatterns](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/servlet/config/annotation/InterceptorRegistration.html#addPathPatterns(java.lang.String...))